### ISSUE TITLE
Parse error with DATE_ADD and literal interval

### ISSUE THREAD

**Issue by barrywhart** [alanmcruickshank/sqlfluff#148]
Input:
```
SELECT
    DATE_ADD(CURRENT_DATE('America/New_York'), INTERVAL 1 year)
```

Results:
```
$ sqlfluff lint test.sql
== [test.sql] FAIL
L:   2 | P:  57 |  PRS | Found unparsable segment @L002P057: '1 year...'
```

**Comment by alanmcruickshank** [comment]
Great example. I've used this as a test case.

**Comment by alanmcruickshank** [comment]
This was also fixed in #155
